libs/shared/contentful/src/lib/queries/eligibility-content-by-plan-ids.ts

```diff
@@ -0,0 +1,117 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Expose, plainToClass, Transform, Type } from 'class-transformer';
+import { IsArray, IsString, MaxLength, ValidateNested } from 'class-validator';
+
+import { graphql } from '../../__generated__/gql';
+
+export const eligibilityContentByPlanIdsQuery = graphql(`
+  query EligibilityContentByPlanIds(
+    $skip: Int!
+    $limit: Int!
+    $locale: String!
+    $stripePlanIds: [String]!
+  ) {
+    purchaseCollection(
+      skip: $skip
+      limit: $limit
+      locale: $locale
+      where: { stripePlanChoices_contains_some: $stripePlanIds }
+    ) {
+      items {
+        stripePlanChoices
+        offering {
+          stripeProductId
+          countries
+          linkedFrom {
+            subGroupCollection(skip: $skip, limit: $limit) {
+              items {
+                groupName
+                offeringCollection(skip: $skip, limit: $limit) {
+                  items {
+                    stripeProductId
+                    countries
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+/**
+ * Transform for the `purchaseCollection` field in the Contentful GraphQL response.
+ */
+function parseEligibilityPurchaseResult({ value }: any) {
+  return (value?.items ?? []).map((value: any) =>
+    plainToClass(EligibilityPurchaseResult, value)
+  );
+}
+
+/**
+ * Transform for the `linkedFrom` field in the Contentful GraphQL response.
+ */
+function parseEligibilitySubGroupResult({ value }: any) {
+  return (value?.subGroupCollection?.items ?? []).map((value: any) =>
+    plainToClass(EligibilitySubGroupResult, value)
+  );
+}
+
+/**
+ * Transform for the `offeringCollection` field in the Contentful GraphQL response.
+ */
+function parseEligibilitySubGroupOfferingResult({ value }: any) {
+  return (value?.items ?? []).map((value: any) =>
+    plainToClass(EligibilitySubGroupOfferingResult, value)
+  );
+}
+
+export class EligibilitySubGroupOfferingResult {
```

Comment:
I haven't dived into this too deeply yet, but I thought the hope with using generated types was that we wouldn't need to manually re-declare the results received from Contentful? (If that's what's happening here)

Reply:
We are asserting these exist. Because the types generated have Maybe's the entire way down, as they don't know if you asked for it or not. This means that everything, everywhere, is optional, even though we want to assert it all exists so that our code isn't a mess. For example, here's the actual query resuult type:
https://github.com/mozilla/fxa/blob/main/libs/shared/contentful/src/__generated__/graphql.ts#L2709

Writing code to utilize this structure in a type-safe manner quickly resulted in constant narrowing/widening of types. Navigating all the other GraphQL bits in how it links the models made it even worse.